### PR TITLE
CEDS-1132 Countries dropdown throughout the journey.

### PR DESCRIPTION
**Pull request by benjaminparker**
Replaces all existing country AutoComplete inputs with accessible component. Also reworks how the country list is supplied to each component, pulling it out of the Controllers and into the views.

### CODE REVIEW

**Comment by PauloPortugal** [comment]
Commits should include the `JIRA id`. `git rebase -i HEAD~8` if you want to change.

File: app/controllers/declaration/DestinationCountriesController.scala [hunk 1]
```diff
@@ -69,14 +61,14 @@ class DestinationCountriesController @Inject()(
     hc: HeaderCarrier
   ): Future[Result] =
     customsCacheService.fetchAndGetEntry[DestinationCountriesSupplementary](cacheId, formId).map {
-      case Some(data) => Ok(destination_countries_supplementary(supplementaryForm.fill(data)))
-      case _          => Ok(destination_countries_supplementary(supplementaryForm))
+      case Some(data) => Ok(destination_countries_supplementary(supplementaryForm.fill(data), Countries.allCountries))
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/DestinationCountriesController.scala]: What do you think to move import to view directly?

File: conf/location-autocomplete-canonical-list.json [hunk 2]
```diff
@@ -1,280 +1,236 @@
...
-  ],
-  [
-    "Andorra",
-    "country:AD"
+    "Armenia",
+    "country:AM"
   ],
   [
-    "Angola",
+    "Angola including Cabinda",
```
Comment (PauloPortugal) [inline conf/location-autocomplete-canonical-list.json]: Wow!
That seems a bit harsh.
This is a massive political statement.
A bit of political context here:
Cabinda is an exclave and province of Angola, surrounded by Democratic Republic of the Congo, they are the richest province of Angola and are responsibles for the majority of oil production.
They are effectively the motor of the Angolan economy and have suggested in the past the will to become independent, although that would probably cause a civil war followed by a war with DRC.

It is the first time I see mentioned `Angola including Cabinda`

File: app/controllers/declaration/DestinationCountriesController.scala (hunk 1 above)
Comment (ed0906) [reply app/controllers/declaration/DestinationCountriesController.scala]: Done

File: test/views/declaration/spec/ViewSpec.scala [hunk 3]
```diff
@@ -41,6 +42,8 @@ trait ViewSpec extends PlaySpec with GuiceOneAppPerSuite with ViewValidator {
   implicit lazy val fakeRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest("", "")
   implicit lazy val messages: Messages = messagesApi.preferred(FakeRequest())
   implicit lazy val flash: Flash = new Flash()
+  
+  
   implicit lazy val countries: List[Country] = Countries.allCountries
```
Comment (PatrykRudnicki) [inline test/views/declaration/spec/ViewSpec.scala]: Probably this line is incorrect. We don't need to pass countries as implicit.

Comment (ed0906) [reply test/views/declaration/spec/ViewSpec.scala]: Removed

**Pull request merged**